app/models/reference.rb

```diff
@@ -29,4 +29,9 @@ def email_address_not_own
 
     errors.add(:email_address, :own) if email_address == candidate_email_address
   end
+
+  def update_token
+    raw_token, _encrypted_token = Devise.token_generator.generate(Reference, :token)
+    update(token: raw_token)
```

Comment:
`update!` to avoid silent failures (we're not checking the return value).

```suggestion
    update!(token: raw_token)
```

Reply:
done